root/utility/tableColumns.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import EntityLink from '../static/scripts/common/components/EntityLink';
+import yesNo from '../static/scripts/common/utility/yesNo';
+
+declare type CellT = {
+  +cell: {
+    +value: any,
+  },
+  +row: {
+    +original: any,
+  },
+};
+
+export function defineNameColumn(
+  defineTitle: () => string,
+) {
+  return {
+    Cell: ({row: {original}}: CellT) => (
+      <EntityLink entity={original} />
+    ),
+    Header: defineTitle,
+    accessor: 'name',
+  };
+}
+
+export const typeColumn = {
+  Cell: ({cell: {value}}: CellT) => l(value),
+  Header: N_l('Type'),
+  accessor: 'typeName',
+  id: 'type',
+};
+
+export const subscriptionColumn = {
+  Cell: ({cell: {value}}: CellT) => yesNo(value),
+  Header: N_l('Subscribed'),
+  accessor: 'subscribed',
+};
+
+export function defineActionsColumn(
+  actions: $ReadOnlyArray<[string, string]>,
+) {
+  return {
+    Cell: ({row: {original}}: CellT) => (
+      <>
+        {actions.map((actionPair, index) => (
+          <React.Fragment key={actionPair[1] + index === 0 ? '-first' : ''}>
```

Comment:
I think the intent here is
```suggestion
          <React.Fragment key={actionPair[1] + (index === 0 ? '-first' : '')}>
```
but that's isn't what the current logic is. Why is `'-first'` needed anyway?

Reply:
Fixed in https://github.com/metabrainz/musicbrainz-server/pull/1246/commits/6ee5d3d424b1dda8aecefe6be1b9f7d2ac4092e1#diff-b9a383135673174ada27c6a8899eba55R47

`-first` stands for _without leading separator_.